src/Till.js

```diff
@@ -0,0 +1,50 @@
+var Till = function() {
+  this.detailsList = hipsterCoffeeDetails;
+  this.orderedItems = [];
+  this.subTotal = 0;
+  this.totalCalculator = new TotalCalculator(this.detailsList, this.orderedItems, this.subTotal);
+};
+
+Till.prototype.retrievePrice = function(item) {
+  return this.detailsList[0].prices[0][item];
+};
+
+Till.prototype.addItem = function(item) {
+  this.subTotal += this.retrievePrice(item);
+  this.orderedItems.push(item);
+  return item;
```

Comment:
Why do you need to return the item? Look up a thing called command/query separation - basically, each function is a command (do something) or a query (what is something?) - this one is both: it modifies the orderedItems array, and then it returns something.